File: leetcodeInterview_17_05/solveLeetcodeInterview_17_05.cpp
// class Solution {

// public:

//     vector<string> findLongestSubarray(vector<string> &array) {

//         int n = array.size(), begin = 0, end = 0, s = n, first[n * 2 + 1];

//         memset(first, -1, sizeof(first));

//         first[s] = 0; // s[0] = 0

//         for (int i = 1; i <= n; ++i) {

//             s += (array[i - 1][0] >> 6 & 1) * 2 - 1;

//             int j = first[s];

//             if (j < 0)

//                 first[s] = i;

//             else if (i - j > end - begin)

//                 begin = j, end = i;

//         }

//         return {array.begin() + begin, array.begin() + end};

//     }

// };

File: interview/os/leetcodeInterview_17_05/solveLeetcodeInterview_17_05.go
// func findLongestSubarray(array []string) []string {
//     n := len(array)
//     begin, end, s := 0, 0, n
//     first := make([]int, n*2+1)

//     for i := range first {
//         first[i] = -1
//     }

//     first[s] = 0

//     for i := 1; i <= n; i++ {
//         s += (int(array[i-1][0]) >> 6 & 1) * 2 - 1
//         j := first[s]

//         if j < 0 {
//             first[s] = i
//         } else if i-j > end-begin {
//             begin, end = j, i
//         }
//     }

//     return array[begin:end]
// }